### DISCUSSION TITLE
Python/cpython zombie `master` branch 🧟‍♂️

### DISCUSSION THREAD

**Discussion by ambv** [community/community#22450] [New to GitHub]
<p>Hi there, we migrated <code>python/cpython</code> from <code>master</code> to <code>main</code> back in May <a href="https://github.community/t/renaming-python-master-branch-to-main-1-4k-prs-700-repositories-triggered-server-http-error-500/178090">with some adventures</a>. It worked ever since until Oct 8th.</p>
<p>Somebody pushed <code>master</code> back to the repository:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/9/5/95ec4077e60c1af905ee1357d49cb9714dccc7e9.png" data-download-href="/uploads/short-url/lohjeOTCaHbKehiqiBiKejkrDrX.png?dl=1" title="Screenshot 2021-10-11 at 23.58.59" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/55281/181070995-23a92225-9b32-4674-a80a-6d8da38d018c.png" alt="Screenshot 2021-10-11 at 23.58.59" data-base62-sha1="lohjeOTCaHbKehiqiBiKejkrDrX" width="690" height="440" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/9/5/95ec4077e60c1af905ee1357d49cb9714dccc7e9_2_690x440.png, https://aws1.discourse-cdn.com/github/optimized/3X/9/5/95ec4077e60c1af905ee1357d49cb9714dccc7e9_2_1035x660.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/9/5/95ec4077e60c1af905ee1357d49cb9714dccc7e9_2_1380x880.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/9/5/95ec4077e60c1af905ee1357d49cb9714dccc7e9_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot 2021-10-11 at 23.58.59</span><span class="informations">2288×1460 460 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>We are sadly unable to delete it as Github rejects the operation, claiming the branch is being renamed right now:</p>
<pre><code class="lang-shell">$ git push upstream --delete master
To github.com:python/cpython.git
 ! [remote rejected]       master (branch master is being renamed)
error: failed to push some refs to 'github.com:python/cpython.git'
</code></pre>
<p>The branch also doesn’t appear in the list of all branches: <a href="https://github.com/python/cpython/branches/all" class="inline-onebox" rel="noopener nofollow ugc">Branches · python/cpython · GitHub</a>, probably for the same reason (it thinks it’s being renamed).</p>
<p>But it can be successfully pulled:</p>
<pre><code class="lang-shell">$ git fetch upstream master
remote: Enumerating objects: 75, done.
remote: Counting objects: 100% (52/52), done.
remote: Total 75 (delta 52), reused 52 (delta 52), pack-reused 23
Unpacking objects: 100% (75/75), 82.45 KiB | 861.00 KiB/s, done.
From github.com:python/cpython
 * branch                  master     -&gt; FETCH_HEAD
   518f8b5dd5..b108db63e0  master     -&gt; upstream/master

$ git checkout --track upstream/master
Branch 'master' set up to track remote branch 'master' from 'upstream'.
Switched to a new branch 'master'
</code></pre>
<p>We’d like it to disappear for good, thank you very much.</p>

**Comment by gpshead** [comment]
<p><a class="mention" href="/aharpole">@aharpole</a> for an attention bump (given you handled <a href="https://github.community/t/renaming-python-master-branch-to-main-1-4k-prs-700-repositories-triggered-server-http-error-500/178090/3" class="inline-onebox">Renaming Python "master" branch to "main" (1.4k PRs &amp; 700 repositories) triggered server HTTP error 500 - #3 by jcgoble3</a>).</p>

**Comment by aharpole** [comment, marked as answer]
<p>Thanks for the report, <a class="mention" href="/gpshead">@gpshead</a>!</p>
<p>There was an extra record of the branch rename that was stuck in the “started” state. I manually marked that one as finished; if you try to delete <code>master</code> again it should permit that. Let me know if you have any further issues with this!</p>

**Comment by ambv** [comment]
<p>Thank you very much, branch successfully deleted!</p>
